frontend/src/pages/sites/siteForm/SiteForm.tsx

```diff
@@ -0,0 +1,157 @@
+import React from "react";
+import { useHistory, Link } from "react-router-dom";
+import { useForm, Controller } from "react-hook-form";
+import { yupResolver } from "@hookform/resolvers/yup";
+import * as yup from "yup";
+import cx from "classnames";
+import { Button, Form } from "react-bootstrap";
+import Select from "react-select";
+
+import { Site_findSite as Site } from "src/graphql/definitions/Site";
+import { ValidSiteTypeEnum, SiteCreateInput } from "src/graphql";
+import { createHref } from "src/utils";
+import { ROUTE_SITES, ROUTE_SITE } from "src/constants/route";
+
+const validSites = Object.keys(ValidSiteTypeEnum);
+
+const schema = yup.object({
+  name: yup.string().required("Name is required"),
+  description: yup.string().optional(),
+  url: yup.string().optional(),
+  regex: yup.string().optional(),
+  valid_types: yup
+    .array(yup.string().oneOf(validSites).required())
+    .min(1, "At least one site type is required")
+    .ensure(),
+});
+
+type SiteFormData = yup.Asserts<typeof schema>;
+
+const capitalizeText = (text: string) =>
+  `${text[0].toUpperCase()}${text.toLowerCase().slice(1)}`;
```

Comment:
You don't have to define this, `lodash` has `capitalize`:
`import { capitalize } from "lodash-es";`